### PR TITLE
Pull Request #17342: add support to enable kinit options [-f|-F] and [-a|-A]

### PR DESCRIPTION
**Pull request by kiwy42**
Feature request #17314
Add to default option in `config.yaml` and `default_airflow.cfg` to cause no regression.
kinit can now emit non forwardable ticket and ticket without originate IP.
Add 2 default configuration value:
[kerberos]
forwardable = True (this one will trigger kinit argument -f or -F)
include_ip = True ( this one will trigger kinit argument -a or -A)

Edit kerberos.py to enable usage of new argument [-f | -F] and [-a | -A]

From old pull request #17342

### CODE REVIEW

**Review by potiuk — COMMENTED** [review]
Looks good, but can you please add integration tests for those flags? We already have kerberos support in Breeze and tests running wiht (you need to start `./breeze --integration kerberos` and the test is security/test_kerberos.py

**Comment by mik-laj** [comment]
Can you add docs for these options?https://airflow.apache.org/docs/apache-airflow/stable/security/kerberos.html

**Review by potiuk — APPROVED** [review]
Looks really nice now!

**Comment by kiwy42** [comment]
> 
> 
> Yep. There was GitHub Packages incident: https://www.githubstatus.com/incidents/wsh4dvm6x81k closed/reopened to rebuild.

OK wasn't sure if it was coming from me or not, however most MSSql integration did fail and I'm pretty sure it's not what I did that influence the issue, is it normal ?

**Comment by potiuk** [comment]
Not your fault for MSSQL (Something we need to fix) but the static check is definitely yours to fix ;).

**Comment by kiwy42** [comment]
> Not your fault for MSSQL (Something we need to fix) but the static check is definitely yours to fix ;).

I'm trying but those pre commit static checks I don''t understand, it compares my file with my previous file but they changed. 
I'm a bit puzzled. 
Also there's those fail that do not create an explicit error:  
```
black....................................................................................Failed
- hook id: black
- files were modified by this hook

reformatted airflow/security/kerberos.py
reformatted tests/security/test_kerberos.py
All done! ✨ 🍰 ✨
2 files reformatted, 2725 files left unchanged.
All done! ✨ 🍰 ✨
244 files left unchanged.
```
and  
```
Checks for consistency between config.yml and default_config.cfg.........................Failed
- hook id: airflow-config-yaml
- files were modified by this hook

Converting ./scripts/ci/pre_commit/../../../airflow/config_templates/config.yml to ./scripts/ci/pre_commit/../../../airflow/config_templates/default_airflow.cfg
Converting ./scripts/ci/pre_commit/../../../airflow/providers/google/config_templates/config.yml to ./scripts/ci/pre_commit/../../../airflow/providers/google/config_templates/default_config.cfg
```
But what Am I suppose to do when it compares to previous code that in fact as changed ? :  
```
diff --git a/tests/security/test_kerberos.py b/tests/security/test_kerberos.py
index 2c1c92e..073d1ef 100644
--- a/tests/security/test_kerberos.py
+++ b/tests/security/test_kerberos.py
@@ -43,43 +43,43 @@ class TestKerberos(unittest.TestCase):
         """
         assert renew_from_kt(principal=self.args.principal, keytab=self.args.keytab) is None
 
-    @conf_vars({('kerberos', 'keytab'): KRB5_KTNAME, ('kerberos','include_ip'): ''})
+    @conf_vars({('kerberos', 'keytab'): KRB5_KTNAME, ('kerberos', 'include_ip'): ''})
```
I did those change to solve formatting checks and it creates more error, I don't understand.  

Thank you for your help.

**Comment by potiuk** [comment]
@kiwy42 - the way how pre-commit works in this case, it "fixes" things for you - but then it fails so that you can manually add those changed files and rerun commit again. Then it should work fine.  Pre-commit will work by default on changes that have been staged in git  (so you run git add on those changese). Most likely it modified your code but then you have not added those changes and re-run pre-commit again (which under the hood would "stash" the changes you've made and re-apply the fix again on the old version).

In short your flow with pre-commit is :

1) You run it
2) In some cases it will apply fixes automatically in others it will tell you that there are things to fix
3) You fix what you need to fix 
4) You run `git add -p` (best) to apply the changes you've done and the changes done by pre-commit
5) You run commit again and observe pre-commit is happy and  green 

J.

**Comment by kiwy42** [comment]
OK, now I'm over 9000 frustrated, and this error: 
```
ERROR: Cannot install apache-airflow-providers-celery==2.0.0.dev0 because these package versions have conflicting dependencies.

The conflict is caused by:
    apache-airflow-providers-celery 2.0.0.dev0 depends on celery>=5.1.2 and ~=5.1
    The user requested (constraint) celery==4.4.7
```  
Is the reason why I stopped trying to build Airflow, dependencies constraints broke so easily, I don't know how or why but this made me stop trying to build anything myself.  
Hope I will one day finish this pull request :D

**Review by ashb — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by ashb** [comment]
Random failures only -- merging!

**Comment by kiwy42** [comment]
Does anyone knows if by anychance this PR will be deploy on next release ?

**Comment by mik-laj** [comment]
Most likely Airflow 2.2. As workaround, you can overwrite file `airflow/security/kerberos.py` after installation. It shouldn't be too much of a maintenance.

**Comment by eladkal** [comment]
It's already available in beta `2.2.0b2`
